.tekton/pipeline/rhtap-staging-e2e.yaml

```diff
@@ -0,0 +1,264 @@
+        - name: deleteExisting
+          value: "true"
+        - name: revision
+          value: "verify-stage"
+      runAfter:
+        - check-merge-label
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      workspaces:
+      - name: output
+        workspace: git
+        subPath: stage-e2e-git
+    - name: stage-verify-e2e-test
+      taskRef:
+        name: stage-verify-e2e-test
+      params:
+      - name: TASK_RETRIES
+        value: "$(context.pipelineTask.retries)"
+      workspaces:
+      - name: stage-e2e-git
+        workspace: git
+        subPath: stage-e2e-git
+      runAfter:
+        - replace-revision
+        - clone-stage-e2e-git
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+    - name: check-rebase-label-2
+      taskRef:
+        name: pr-rebase-labels
+      params:
+        - name: pull_request_number
+          value: $(params.pull_request_number)
+        - name: match_label
+          value: "needs-rebase"
+      runAfter:
+        - stage-verify-e2e-test
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      workspaces:
+        - name: shared-file-path
+          workspace: shared-file
+    - name: remove-merge-label
+      taskRef:
+        name: remove-pr-labels
+      params:
+        - name: match_label
+          value: "ok-to-merge"
+        - name: pull_request_number
+          value: $(params.pull_request_number)
+      when:
+        - input: $(tasks.check-rebase-label-2.results.matched-rebase-label)
+          operator: in
+          values: ["True"]
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      workspaces:
+        - name: shared-file-path
+          workspace: shared-file
+      runAfter:
+        - check-rebase-label-2
+    - name: rebase-pr-3
+      taskRef:
+        name: rebase-pr
+      params:
+        - name: pull_request_number
+          value: $(params.pull_request_number)
+        - name: source_branch
+          value: $(params.source_branch)
+      workspaces:
+        - name: shared-file-path
+          workspace: shared-file
+      runAfter:
+        - remove-merge-label
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+        - input: $(tasks.check-rebase-label-2.results.matched-rebase-label)
+          operator: in
+          values: ["True"]
+  finally:
+    - name: revert-replace-revision
+      taskRef:
+        name: revert-replace-revision
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+    - name: merge-pr
```

Comment:
Why is this one in the finally and not `run-after: rebase-pr-post-test`?

That would simplify the task definition as the ` when` could be removed.

Reply:
Not the whole `when` can be removed. Although I have kept it inside finally because I wanted to revert the change of argocd component to its previous deployment before the pr is merged. So in this case finally should execute the `revert-replace-revision` task as well.

Reply:
Why not? This task is `merge-pr` and should be executed only if everything else before it succeeded.

I agree that `revert-replace-revision` should always be run.

Maybe we can talk about it at our next standup.